cmd/ipfs/daemon.go

```diff
@@ -205,6 +209,42 @@ func defaultMux(path string) corehttp.ServeOption {
 	}
 }
 
+// setMemoryLimit a soft memory limit to enforce running the GC more often when
+// we are about to run out.
+// This allows to recoop memory when it's about to run out and cancel the
+// doubled memory footprint most go programs experience, at the cost of more CPU
+// usage in memory tight conditions. This does not increase CPU usage when memory
+// is plenty available, it will use more CPU and continue to run in cases where Kubo
+// would OOM.
+func setMemoryLimit() {
+	// From the STD documentation:
+	// A negative input does not adjust the limit, and allows for retrieval of the currently set memory limit.
+	if currentMemoryLimit := debug.SetMemoryLimit(-1); currentMemoryLimit != math.MaxInt64 {
+		fmt.Printf("GOMEMLIMIT already set to %s, leaving as-is.\n", humanize.IBytes(uint64(currentMemoryLimit)))
```

Comment:
shall we use `log` here?

Reply:
No, all of this file use `fmt` for non erroring output https://github.com/ipfs/kubo/blob/6b3e242f30f744c4a38f2a89e09e8df42f8b1b9c/cmd/ipfs/daemon.go#L784.